src/ExpressionEvaluator/Core/Source/ExpressionCompiler/CustomTypeInfo.cs

```diff
@@ -167,41 +155,53 @@ private static ReadOnlyCollection<string> DecodeNames(ReadOnlyCollection<byte> b
             int length = bytes.Count - start;
             var array = CopyBytes(bytes, start, length);
```

Comment:
> var array = CopyBytes(bytes, start, length); [](start = 11, length = 45)

This file is inconsistent on whether or not it cares about array pooling.  Some methods use it, others don't.  This is a place where pooling would be possible it seems.

Is there a reason to the inconsistency?